lib/std/os.zig

```diff
@@ -2822,66 +2835,78 @@ pub const AcceptError = error{
 pub fn accept(
     /// This argument is a socket that has been created with `socket`, bound to a local address
     /// with `bind`, and is listening for connections after a `listen`.
-    sockfd: fd_t,
+    sock: socket_t,
     /// This argument is a pointer to a sockaddr structure.  This structure is filled in with  the
     /// address  of  the  peer  socket, as known to the communications layer.  The exact format of the
     /// address returned addr is determined by the socket's address  family  (see  `socket`  and  the
     /// respective  protocol  man  pages).
-    addr: *sockaddr,
+    addr: ?*sockaddr,
     /// This argument is a value-result argument: the caller must initialize it to contain  the
     /// size (in bytes) of the structure pointed to by addr; on return it will contain the actual size
     /// of the peer address.
     ///
     /// The returned address is truncated if the buffer provided is too small; in this  case,  `addr_size`
     /// will return a value greater than was supplied to the call.
-    addr_size: *socklen_t,
+    addr_size: ?*socklen_t,
     /// The following values can be bitwise ORed in flags to obtain different behavior:
     /// * `SOCK_NONBLOCK` - Set the `O_NONBLOCK` file status flag on the open file description (see `open`)
     ///   referred  to by the new file descriptor.  Using this flag saves extra calls to `fcntl` to achieve
     ///   the same result.
     /// * `SOCK_CLOEXEC`  - Set the close-on-exec (`FD_CLOEXEC`) flag on the new file descriptor.   See  the
     ///   description  of the `O_CLOEXEC` flag in `open` for reasons why this may be useful.
     flags: u32,
-) AcceptError!fd_t {
-    const have_accept4 = comptime !std.Target.current.isDarwin();
+) AcceptError!socket_t {
+    const have_accept4 = comptime !(std.Target.current.isDarwin() or builtin.os.tag == .windows);
```

Comment:
Should be able to use `AcceptEx`?

Reply:
Maybe. This one, right? https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-acceptex
I don't immediately see the advantage of using that one. And there seems to be one big disadvantage, which is that you need to obtain the function pointer to the `AcceptEx` function at runtime. (Though this could go in a initialize function together with the `WSAStartup`.)

Reply:
> there seems to be one big disadvantage, which is that you need to obtain the function pointer to the `AcceptEx` function at runtime. (Though this could go in a initialize function together with the `WSAStartup`.)

ouch.

In that case what you've got is fine.

Unless you want to end-game it and go for AFD :)

Reply:
Let's take small steps and leave it like this for now ;)